### ISSUE TITLE
v2 SMS Validation

### ISSUE THREAD

**Issue by k-macmillan** [department-of-veterans-affairs/va-enp-api#112]
## User Story - Business Need
Validate `/legacy/v2/notifications/sms` in the same manner notification-api validates them. Evaluating which follow-on tasks should be "chained" is not necessary.

- [x] Sync with Kyle when ticket is picked up.
- [x] Ticket is understood, and QA has been contacted (if the ticket has a QA label).

### User Story(ies)

**As a** VA Notify client
**I want** any upgrades to be seamless for me
**So that** I don't have to change for any VA Notify changes

### Additional Info and Resources

Starts with `post_notification` in the notification-api.

- Validate (*skip* = not doing, *defer* = doing later)
	1. Template is of the right channel type: sms and is active
	2. Personalization matches template personalization (permissive, needs them if there, extra is ignored)
	3. *defer*: template is part of the service - will come when we have proper JWT auth and request state
	4. *defer*: rate limiting - will come later (only applies to dev/staging)
	5. *defer*: daily message limiting - will come later (only applies to dev/staging)
	6. *skip*: service has permission to send this channel of notification
	7. *skip*: template is active - no longer applicable
	8. *skip*: scheduled for - unused
	9. *skip*: sms sender default - all services have a default sms sender now

## Acceptance Criteria

- [x] SOLID principles
- [x] route entrypoint calls a validate method
- [x] All non-skipped/deffered methods validations in the Additional Info have methods written
- [x] Other than identifying what to validate, no notification-api code is reused
- [x] All validations involving the database are done against the notification-api read database
- [x] This work is added to the sprint review slide deck (key win bullet point and demo slide)

## QA Considerations

Endpoint log parity or identification of what will change so we can notify our clients (when we get far enough along to switch).

## Potential Dependencies

- Definitely depends on #114 
- Pydantic Models -#117

## Out of Scope
- Evaluating chain tasks.
- Sending to Celery/SQS.

**Comment by cris-oddball** [comment]
@k-macmillan potentially dumb question, but does ENP have permissions to read from the notification-api database? Or do we need a ticket for that (upon which this ticket is dependent)?

**Comment by cris-oddball** [comment]
Nevermind, I'll add #114 to dependencies

**Comment by cris-oddball** [comment]
### QA CONVO
via Slack

- The only two non-deferred or skipped validations are that template is of type sms and that personalisation is handled and permissive
- Please update QA if anything changes.

**Comment by EvanParish** [comment]
I've made good progress so far and I've been trying to stick to a TDD approach when adding this validation. Unfortunately adding this is taking a bit longer than expected, especially when fighting with unit tests that don't seem to want to mock out the database calls correctly at times.

**Comment by EvanParish** [comment]
I finished wrapping up all the testing yesterday and opened the PR for review at the end of the day, just waiting on reviews now.

**Comment by cris-oddball** [comment]
PR approved and merged, will deploy to Perf and begin validations soon.

**Comment by cris-oddball** [comment]
### QA TESTING IN PROGRESS

- [ ] TC01 Happy path: all properties in payload, valid template_id, should return a 201 and any personalisation passed. Logs provide the payload received and the response sent. The response includes fixed properties as we are not writing to the napi DB but the personalisation passed is shown in the response.

Logs
![Image](https://github.com/user-attachments/assets/053240d1-379d-4e82-b941-ae482781f806)

Payload & Response

![Image](https://github.com/user-attachments/assets/f56eb373-beac-4967-b057-524cd0717218)

- [ ] TC02 Negative: non-existent template_id in payload, expect 400 error in notification-api style error message
- [ ] TC03 Negative: email template_id used in payload, expect 400 error in notification-api style error message
- [ ] TC04 Negative: archived sms template used in payload, expect 400 error in notification-api style error message
- [ ] TC05 Negative: personalisation in payload does not match template personalisation, expect 400 error in notification-api style error message
- [ ] TC06 Negative: personalisation in payload is not an object of type dictionary, expect 400 error in notification-api style error message

Note that properties other than the ones in this ticket were not tested.

Logs/Responses supporting Negative test cases:

**Comment by cris-oddball** [comment]
Bug ticket created ^
